src/main/kotlin/lifeCycle/LifeCycle.kt

```diff
@@ -0,0 +1,32 @@
+package lifeCycle
+
+class LifeCycle(state: LifeCycleState) {
+    private var lifeCycleListeners = mutableListOf<LifeCycleListener>()
+
+    var state: LifeCycleState = state
+        private set
+
+    fun changeState(state: LifeCycleState) {
+        this.state = state
+        lifeCycleListeners.forEach { it.onStateChange(state) }
+        if (state == LifeCycleState.DESTROYED)
+            lifeCycleListeners.clear()
+    }
+
+    fun isChangeAble(): Boolean {
+        return when (state) {
+            LifeCycleState.LIVE -> true
+            else -> false
+        }
+    }
+
+    fun getLifeCycleListeners() = lifeCycleListeners.toList()
+
+    fun addLifeCycleListener(onStateChanged: LifeCycleListener) {
+        lifeCycleListeners.add(onStateChanged)
+    }
+
+    fun interface LifeCycleListener {
```

Comment:
```suggestion
    fun interface Listener Or OnChangeListener {
```
nested interface or class 사용시 naming convention 또한 nested하게 작성하는 습관을 들이는게 좋습니다 😄